### ISSUE TITLE
googleapi: Error 400: Invalid value for field 'resource.network': 'projects/b/global/networks/c'. Network field cannot be modified., invalid

### ISSUE THREAD

**Issue by chandra-scentre** [terraform-google-modules/terraform-google-lb-internal#152]
### TL;DR

Have a Google Internal LB which got the backend.. it was created some 5 years back and now when i am trying to update the module version 

Error when applying the module

Error updating RegionBackendService "projects/abc/regions/us-east1/backendServices/hc": googleapi: Error 400: Invalid value for field 'resource.network': 'projects/b/global/networks/c'. Network field cannot be modified., invalid

### Observed behavior

  # module.abc.module.ilb.google_compute_region_backend_service.default will be updated in-place
  ~ resource "google_compute_region_backend_service" "default" {
      ~ connection_draining_timeout_sec = 0 -> 300
        id                              = "projects/abc/regions/us-east1backendServices/ilb-with-http-hc"
        name                            = "http-hc"
      + network                         = "https://www.googleapis.com/compute/v1/projects/xyzglobal/networks/abc"
        # (17 unchanged attributes hidden)

        # (3 unchanged blocks hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

### Terraform Configuration

```hcl
module "ilb" {
  source          = "GoogleCloudPlatform/lb-internal/google"
  version         = "7.0.0"
  name            = "ilb"
  region          = var.region
  ip_address      = google_compute_address.ilb.address
  ip_protocol     = "TCP"
  ports           = []
  all_ports       = true
  network         = var.host_network_name
  network_project = var.host_network_project_id
  subnetwork      = var.subnet_name
  health_check    = local.health_check
  source_tags     = ["${var.gce_prefix}-ingress"]
  target_tags     = [var.gce_prefix]
  backends = [
    { group = module.mig-a.instance_group, description = "worker-a" },
    { group = module.mig-b.instance_group, description = "worker-b" },
    { group = module.mig-c.instance_group, description = "worker-c" },
  ]
  service_label = "test"
}
```

### Terraform Version

```sh
1.9.8
```

### Additional information

when running the google describe command (note changed some project names and regions)

main thing to observe is that it is created in 2020 and there is no network in this describe .. now terraform module is trying to add the network module

gcloud compute backend-services describe http-hc \
    --region=us-east1 \
    --project=abc
backends:
- balancingMode: CONNECTION
  description: worker-b
  group: https://www.googleapis.com/compute/v1/projects/def/regions/us-east1/instanceGroups/mig-b
- balancingMode: CONNECTION
  description: worker-a
  group: https://www.googleapis.com/compute/v1/projects/def/regions/us-east1/instanceGroups/mig-a
- balancingMode: CONNECTION
  description: worker-c
  group: https://www.googleapis.com/compute/v1/projects/def/regions/us-east1/instanceGroups/mig-c
connectionDraining:
  drainingTimeoutSec: 0
creationTimestamp: '2020-06-16T18:36:48.137-07:00'
description: ''
fingerprint: P1O2dKi-aWI=
healthChecks:
- https://www.googleapis.com/compute/v1/projects/abc/global/healthChecks/hc
id: 'mmmmm'
kind: compute#backendService
loadBalancingScheme: INTERNAL
name: http-hc
protocol: TCP
region: https://www.googleapis.com/compute/v1/projects/abc/regions/us-east1
selfLink: https://www.googleapis.com/compute/v1/projects/abc/regions/us-east1/backendServices/http-hc
sessionAffinity: NONE
timeoutSec: 10

**Comment by mambor177** [comment]
Has anyone found a solution to this?  I have backend that were created years ago failing with the same error message.

**Comment by BapRx** [comment]
The last working tag for ILB created using older versions of the module is `4.4.1` since the network argument was added in https://github.com/terraform-google-modules/terraform-google-lb-internal/pull/91

It seems like the API doesn't return the network value at all although it's visible in GCP console. Manually editing the Terraform state doesn't solve anything because the API replaces the value with `""`.

I also tried `terraform state rm && terraform import` without luck.
